Kernel/CMakeLists.txt

```diff
@@ -362,6 +365,32 @@ endif()
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pie -Wl,--no-dynamic-linker")
 
+# Kernel Coverage (KCOV) is an API to collect and expose program counters of
+# kernel code that has been run to user space. It's rather slow and likely not
+# secure to run in production builds. Useful for coverage guided fuzzing.
+if (ENABLE_KERNEL_COVERAGE_COLLECTION)
+    add_definitions(-DENABLE_KERNEL_COVERAGE_COLLECTION)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize-coverage=trace-pc")
+    # Make sure we don't instrument any code called from __sanitizer_cov_trace_pc
+    # otherwise we'll end up with recursive calls to that function.
+    set_source_files_properties(../AK/Format.cpp PROPERTIES COMPILE_FLAGS "-fno-sanitize-coverage=trace-pc")
+    set_source_files_properties(../AK/StringBuilder.cpp PROPERTIES COMPILE_FLAGS "-fno-sanitize-coverage=trace-pc")
+    set_source_files_properties(../Kernel/Arch/x86/${KERNEL_ARCH}/Processor.cpp PROPERTIES COMPILE_FLAGS "-fno-sanitize-coverage=trace-pc")
+    set_source_files_properties(../Kernel/Devices/KCOVDevice.cpp PROPERTIES COMPILE_FLAGS "-fno-sanitize-coverage=trace-pc")
+    set_source_files_properties(../Kernel/Devices/KCOVInstance.cpp PROPERTIES COMPILE_FLAGS "-fno-sanitize-coverage=trace-pc")
+    set_source_files_properties(../Kernel/FileSystem/File.cpp PROPERTIES COMPILE_FLAGS "-fno-sanitize-coverage=trace-pc")
+    set_source_files_properties(../Kernel/FileSystem/FileDescription.cpp PROPERTIES COMPILE_FLAGS "-fno-sanitize-coverage=trace-pc")
+    set_source_files_properties(../Kernel/Heap/SlabAllocator.cpp PROPERTIES COMPILE_FLAGS "-fno-sanitize-coverage=trace-pc")
+    set_source_files_properties(../Kernel/init.cpp PROPERTIES COMPILE_FLAGS "-fno-sanitize-coverage=trace-pc")
+    set_source_files_properties(../Kernel/SanCov.cpp PROPERTIES COMPILE_FLAGS "-fno-sanitize-coverage=trace-pc")
+    # GCC assumes that the caller saves registers for functions according to the
+    # System V ABI and happily inserts coverage calls into the function prologue
+    # for all functions. This assumption is not true for interrupt handlers
+    # because their calling convention is not compatible with the System V ABI.
+    set_source_files_properties(../Kernel/Arch/x86/common/Interrupts.cpp PROPERTIES COMPILE_FLAGS "-fno-sanitize-coverage=trace-pc")
+    set_source_files_properties(../Kernel/Syscall.cpp PROPERTIES COMPILE_FLAGS "-fno-sanitize-coverage=trace-pc")
```

Comment:
Maybe add a variable for those so we don't have to repeat the option:

```CMake
set(KCOV_EXCLUDED_SOURCES ...)
set_source_files_properties(${KCOV_EXCLUDED_SOURCES} PROPERTIES ...)
```

Reply:
makes sense